src/main/java/org/opentripplanner/middleware/triptracker/TravelerLocator.java

```diff
@@ -86,6 +91,12 @@ public static String getInstruction(
                 }
             }
         }
+
+        try {
+            new CheckMonitoredTrip(travelerPosition.trip).checkForMajorTripChanges(travelerPosition.nextLeg);
+        } catch (Exception e) {
+            LOG.error("Error encountered while checking for major trip changes.", e);
+        }
```

Comment:
When a traveler approaches a bus leg from a walk leg, this block doesn't get called. In the case of a walk->bus transition, the block at line 67 is executed instead. The code in this block needs to execute with the logic when arriving at the end of a leg where the next leg is on a different mode.